packages/relay/src/lib/eth.ts

```diff
@@ -1109,69 +1110,96 @@ export class EthImpl implements Eth {
-          throw predefined.INTERNAL_ERROR();
-        }
+    if (e instanceof SDKClientError) {
+      this.hapiService.decrementErrorCounter(e.statusCode);
+    }
 
-        if (record.ethereumHash == null) {
-          this.logger.error(`${requestIdPrefix} The ethereumHash can never be null for an ethereum transaction, and yet it was!!`);
-          throw predefined.INTERNAL_ERROR();
-        }
+    if (!txSubmitted) {
+      return predefined.INTERNAL_ERROR(e.message.toString());
+    }
 
-        return  EthImpl.prepend0x(Buffer.from(record.ethereumHash).toString('hex'));
-      } catch (e) {
+    await this.mirrorNodeClient.getContractRevertReasonFromTransaction(e, requestId, requestIdPrefix);
 
-        await this.mirrorNodeClient.getContractRevertReasonFromTransaction(e, requestId, requestIdPrefix);
+    this.logger.error(e,
+        `${requestIdPrefix} Failed sendRawTransaction during record retrieval for transaction ${transaction}, returning computed hash`);
+    //Return computed hash if unable to retrieve EthereumHash from record due to error
+    return EthImpl.prepend0x(createHash('keccak256').update(transactionBuffer).digest('hex'));
+  }
 
-        this.logger.error(e,
-          `${requestIdPrefix} Failed sendRawTransaction during record retrieval for transaction ${transaction}, returning computed hash`);
-        //Return computed hash if unable to retrieve EthereumHash from record due to error
-        return EthImpl.prepend0x(createHash('keccak256').update(transactionBuffer).digest('hex'));
-      }
-    } catch (e: any) {
-      this.logger.error(e,
-        `${requestIdPrefix} Failed to successfully submit sendRawTransaction for transaction ${transaction}`);
-      if (e instanceof JsonRpcError) {
-        return e;
+  /**
+   * Submits a transaction to the network for execution.
+   *
+   * @param transaction
+   */
+  async sendRawTransaction(transaction: string, requestId?: string): Promise<string | JsonRpcError> {
+    const requestIdPrefix = formatRequestIdMessage(requestId);
+    if(transaction?.length >= constants.FUNCTION_SELECTOR_CHAR_LENGTH)
+      this.ethExecutionsCounter.labels(EthImpl.ethSendRawTransaction, transaction.substring(0, constants.FUNCTION_SELECTOR_CHAR_LENGTH)).inc();
+
+    const parsedTx = await this.parseRawTxAndPrecheck(transaction, requestId);
+    const transactionBuffer = Buffer.from(EthImpl.prune0x(transaction), 'hex');
+
+    let txSubmitted = false;
+    try {
+      const contractExecuteResponse = await this.hapiService.getSDKClient().submitEthereumTransaction(transactionBuffer, EthImpl.ethSendRawTransaction, requestId);
+      txSubmitted = true;
+      // Wait for the record from the execution.
+      let txId = contractExecuteResponse.transactionId.toString();
+      const formattedId = formatTransactionId(txId);
+      const  record = await this.mirrorNodeClient.repeatedRequest('getContractResult', [formattedId], 3, requestId);
```

Comment:
nit: `'getContractResult'` should be a const noted in MirrorNodeClient that matches the API name it exposes

Reply:
`getContractResult` is a method name, that will corresponds to `contracts/results/` so its  value should match the name of the method to work. and setting the Const name as contractsResults o `CONTRACTS_RESULTS_METHOD_NAME` does not seem to describe the purpose of the parameter.

Although I agree on referencing instead of hardcoding the value, so I have got the name from the function that is being called itself.

let me know your thoughts.